### PR TITLE
Milestone #2: Add and remove item implemented

### PR DESCRIPTION
**Pull request by lordksix**
In this PR, the charges are as follows:
- Implement Add Item functionality
- Implement remove edit item functionality
- Implement item status
- Save information in local storage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,49 +1,80 @@
...
+  if (itemCompleted.length > 0) {
+    itemCompleted.forEach((delItem) => {
+      ListOfItems.removeitem(parseInt(delItem.dataset.index, 10), delItem, listUl);
+    });
+    itemCompleted.forEach((delItem) => {
+      listUl.removeChild(delItem);
+    });
+    ListOfItems.updateList('.app-item');
   }
 };
```
Comment (tresorsawasawa) [inline src/index.js]: - [x] As required for the project, kindly make sure the `Delete` task functionality is working as expected. Right now, User cannot delete a task item.

- [x] And once a User has deleted a task, kindly make sure the **Deleting**  action should **update all remaining items' indexes**, so they represent the current list order and are unique and the changes are saved in `local storage`.

File: src/modules/ItemList.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  updateList(itemClass) {
+    const itemsRender = document.querySelectorAll(itemClass);
+    this.itemArray.forEach((listitem, i) => {
+      const id = i + 1;
+      listitem.index = id;
+      itemsRender[i].dataset.index = id;
+    });
+    localStorage.setItem(this.localName, JSON.stringify(this.itemArray));
+  }
+
```
Comment (tresorsawasawa) [inline src/modules/ItemList.js]: - [x] Good job trying to add the `Editing` functionality to your project. However I noticed that the `editing` action doesn't update the Local storage data as it is required for the project. Right now, after deleting a task description and reload the page, your UI display the olde data.

- [x] In short, kindly make sure all changes to the To-Do List are correctly saved in local storage(i.e. Adding a task item, Removing a task item, editing a task description,...) for the project  requirements.

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @lordksix 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/ItemList.js (hunk 2 above)
Comment (lordksix) [reply src/modules/ItemList.js]: Fix edit functionality (all edit are live with local storage) with commit cd2cc9bb48a48e65d5067e61f2397f98fca24f61

File: src/index.js (hunk 1 above)
Comment (lordksix) [reply src/index.js]: I add a single item removal functionality, no matter if it is completed or not, in commit cf8daae1122d8d195e7a9839c8f61942ba1ba186

An update of indexes was done time an item was removed from the list of items:
https://github.com/lordksix/Minimalistic-ToDoList-App/blob/03608d80bbd4a4c2d57aef6648dd2ffd8c7ba81f/src/index.js#L87
https://github.com/lordksix/Minimalistic-ToDoList-App/blob/7d24fa86baefb479dda485e132a2019f758ec47a/src/modules/ItemList.js#L87-L100

**Review by Ibizugbe — APPROVED** [review]
Hi @lordksix 👋👋👋, 

VeryNiceNiceGIF


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅✅

- [x] Linters are passing👍.
- [x] Delete functionality works properly👍.
- [x] Edit functionality works properly and it's persistent to local storage👍.
- [x] Commit messages are descriptive and meaningful👍.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**